dotCMS/src/main/java/com/dotcms/content/elasticsearch/business/ESContentletAPIImpl.java

```diff
@@ -4987,352 +4929,479 @@ private Contentlet internalCheckin(Contentlet contentlet,
 
-                        // publish/expire dates changed
-                        APILocator.getIdentifierAPI().save(ident);
+                    // publish/expire dates changed
+                    APILocator.getIdentifierAPI().save(ident);
 
-                        // we take all inodes associated with that identifier
-                        // remove them from cache and then reindex them
-                        final List<ContentletVersionInfo> list = APILocator.getVersionableAPI()
-                                .findContentletVersionInfos(ident.getId());
+                    // we take all inodes associated with that identifier
+                    // remove them from cache and then reindex them
+                    final List<ContentletVersionInfo> list = APILocator.getVersionableAPI()
+                            .findContentletVersionInfos(ident.getId());
 
-                        final List<String> inodes = new ArrayList<>();
-                        for(final ContentletVersionInfo cvi : list) {
-                            inodes.add(cvi.getWorkingInode());
-                            if(UtilMethods.isSet(cvi.getLiveInode()) && !cvi.getWorkingInode().equals(cvi.getLiveInode())){
-                                inodes.add(cvi.getLiveInode());
-                            }
-                        }
-                        for(final String inode : inodes) {
-                            CacheLocator.getContentletCache().remove(inode);
-                            final Contentlet ct = APILocator.getContentletAPI().find(inode, sysuser, false);
-                            APILocator.getContentletIndexAPI().addContentToIndex(ct,false);
+                    final List<String> inodes = new ArrayList<>();
+                    for(final ContentletVersionInfo cvi : list) {
+                        inodes.add(cvi.getWorkingInode());
+                        if(UtilMethods.isSet(cvi.getLiveInode()) && !cvi.getWorkingInode().equals(cvi.getLiveInode())){
+                            inodes.add(cvi.getLiveInode());
                         }
                     }
+                    for(final String inode : inodes) {
+                        CacheLocator.getContentletCache().remove(inode);
+                        final Contentlet ct = APILocator.getContentletAPI().find(inode, systemUser, false);
+                        APILocator.getContentletIndexAPI().addContentToIndex(ct,false);
+                    }
                 }
             }
+        }
+    }
 
-            final Structure hostStructure = CacheLocator.getContentTypeCache().getStructureByVelocityVarName("Host");
-            if ((contentlet != null) && InodeUtils.isSet(contentlet.getIdentifier()) && contentlet.getStructureInode().equals(hostStructure.getInode())) {
-                final HostAPI hostAPI = APILocator.getHostAPI();
-                hostAPI.updateCache(new Host(contentlet));
+    private void relateTags(Contentlet contentlet, HashMap<String, String> tagsValues,
```

Comment:
void returns are a code smell.   I don't think this method should mutate the content state.  Shouldn't this return a new content object and not possibly poison cache?

Reply:
I think it's fine. I intend to update the reference passed down from internal-checkin here adding the respective tags 
Now regarding the void returns I think is not a bad practice 
https://stackoverflow.com/questions/52059753/is-mutating-object-parameters-in-a-methodin-java-a-bad-practice#52059997
But I'm changing the method so the original content is also returned.